bookkeeper-server/src/main/java/org/apache/bookkeeper/client/LedgerHandle.java

```diff
@@ -787,6 +788,35 @@ public void onFailure(Throwable cause) {
         }
     }
 
+    /*
+     * Read the last entry in the ledger
+     *
+     * @param cb
+     *            object implementing read callback interface
+     * @param ctx
+     *            control object
+     */
+    public void asyncReadLastEntry(ReadCallback cb, Object ctx) {
+        long lastEntryId = getLastAddConfirmed();
+        if (lastEntryId < 0) {
+            // Ledger was empty, so there is no last entry to read
+            cb.readComplete(BKException.Code.NoSuchEntryException, this, null, ctx);
+        } else {
+            asyncReadEntriesInternal(lastEntryId, lastEntryId, cb, ctx);
+        }
+    }
+
+    public org.apache.bookkeeper.client.api.LedgerEntry readLastEntry()
+        throws InterruptedException, ExecutionException, BKException {
```

Comment:
we should not throw "ExcuctionException" for sync methods. keep the behaviour same as other sync methods.

you can check https://github.com/apache/bookkeeper/blob/master/bookkeeper-server/src/main/java/org/apache/bookkeeper/client/LedgerHandle.java#L594 as an example.

you can use SyncReadCallback as other methods are using.